tests/test_presence_do.py

```
@@ -0,0 +1,494 @@
+        att2 = {"session_id": "s2", "user_id": "alice", "display_name": "Alice",
+                "x": 0.9, "y": 0.9, "emoji": "", "hand_raised": True}
+        ws1, ws2 = _make_ws(att1), _make_ws(att2)
+        ctx = _make_ctx(sockets=[ws1, ws2])
+        do  = worker.PresenceDO(ctx, _make_presence_env())
+
+        # First socket wins for the presence record
+        assert do.presence["alice"]["x"] == pytest.approx(0.1)
+
+
+# ===========================================================================
+# 2.  on_fetch – WebSocket upgrade
```

Ruff (0.15.10)

[warning] Comment contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)